### PR TITLE
Convert vAelin to AELIN

### PR DESCRIPTION
**Pull request by saeta-eth**

### CODE REVIEW

File: containers/ContractsInterface/contracts/VAelinConverter.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+			},
+		],
+		stateMutability: 'view',
+		type: 'function',
+	},
+];
+
+const contract = {
+	10: {
+		// TODO: Use OP VAelinConverter address
```
Comment (saeta-eth) [inline containers/ContractsInterface/contracts/VAelinConverter.ts:196]: TODO

File: containers/ContractsInterface/contracts/vAelinToken.ts [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+			},
+		],
+		stateMutability: 'nonpayable',
+		type: 'function',
+	},
+];
+
+const contract = {
+	10: {
+		// TODO: Use OP VAelinConverter address
```
Comment (saeta-eth) [inline containers/ContractsInterface/contracts/vAelinToken.ts:287]: TODO

File: sections/Claim/ConvertSection/ConvertSection.tsx [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+		useState<GasLimitEstimate>(null);
+
+	const { gasPrice, setGasPrice, setTxState } = TransactionData.useContainer();
+	const { monitorTransaction } = TransactionNotifier.useContainer();
+	const { walletAddress, network, signer } = Connector.useContainer();
+
+	const vAelinConverterContract = useMemo(() => {
+		if (!signer || !network?.id) return null;
+
+		const vAelinConverterContract = (getKeyValue(VAelinConverterContract) as any)(
```
Comment (0xcdb) [inline sections/Claim/ConvertSection/ConvertSection.tsx:40]: can you please use `ContractsInterface` for this contract initialisation? I mentioned it the other day.

File: sections/Claim/ConvertSection/ConvertSection.tsx [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+		if (!vAelinConverterContract) return null;
+
+		return new ethers.Contract(
+			vAelinConverterContract.address,
+			vAelinConverterContract.abi,
+			signer
+		);
+	}, [signer, network?.id]);
+
+	const vAelinTokenContract = useMemo(() => {
```
Comment (0xcdb) [inline sections/Claim/ConvertSection/ConvertSection.tsx:53]: Same here.

File: sections/Claim/ConvertSection/ConvertSection.tsx [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+				vAelinConverterContract.address
+			);
+			setHasAllowance(!!Number(ethers.utils.formatEther(allowance)));
+		} catch (e) {
+			console.log(e);
+			setHasAllowance(false);
+		}
+	}, [vAelinTokenContract, vAelinConverterContract, walletAddress]);
+
+	const getVAelinBalance = useCallback(async () => {
```
Comment (0xcdb) [inline sections/Claim/ConvertSection/ConvertSection.tsx:102]: Can you please have a look at what I did for the staking contract? You can use a query I did for this. `useGetTokenBalance`. All you have to do is pass a contract address to the query. Please do not reinvent the wheel and make sure to reuse all the existing components/queries if possible.

File: sections/Claim/ConvertSection/ConvertSection.tsx [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+		try {
+			const tx = await vAelinTokenContract.balanceOf(walletAddress);
+			setHasVAelinBalance(!!Number(ethers.utils.formatEther(tx)));
+		} catch (e) {
+			console.log(e);
+			setHasVAelinBalance(false);
+		}
+	}, [vAelinTokenContract, walletAddress]);
+
+	const handleConverAll = useCallback(async () => {
```
Comment (0xcdb) [inline sections/Claim/ConvertSection/ConvertSection.tsx:114]: Typo. Can you install a spell checker in your IDE? It helps a lot trust me.

**Review by 0xcdb — CHANGES_REQUESTED** [review]
It looks very good. But you are reinventing the wheel a bit everywhere. This file can be simplified if you follow my comments :)

File: sections/Claim/ConvertSection/ConvertSection.tsx (hunk 3 above)
Comment (saeta-eth) [reply sections/Claim/ConvertSection/ConvertSection.tsx]: Done

File: sections/Claim/ConvertSection/ConvertSection.tsx (hunk 4 above)
Comment (saeta-eth) [reply sections/Claim/ConvertSection/ConvertSection.tsx]: Done

File: sections/Claim/ConvertSection/ConvertSection.tsx (hunk 5 above)
Comment (saeta-eth) [reply sections/Claim/ConvertSection/ConvertSection.tsx]: Done

File: sections/Claim/ConvertSection/ConvertSection.tsx (hunk 6 above)
Comment (saeta-eth) [reply sections/Claim/ConvertSection/ConvertSection.tsx]: Done

**Review by 0xcdb — APPROVED** [review]

**Review by Aelin-xyz — APPROVED** [review]

**Pull request merged**